lib/Controller/OpenIdConnectUserInfoController.php

```diff
@@ -99,4 +116,33 @@ private function getUser(AccessTokenEntity $accessToken)
 
         return $user;
     }
+
+    /**
+     * Handle CORS 'preflight' requests by checking if 'origin' is registered as allowed to make HTTP CORS requests,
+     * typically initiated in browser by JavaScript clients.
+     * @param ServerRequest $request
+     * @return Response
+     * @throws OidcServerException
+     */
+    protected function handleCors(ServerRequest $request): Response
+    {
+        $origin = $request->getHeaderLine('Origin');
+
+        if (empty($origin)) {
+            throw OidcServerException::requestNotSupported('CORS error: no Origin header present');
+        }
+
+        if (! $this->allowedOriginRepository->has($origin)) {
```

Comment:
so do clients not send their authorization token as a header when they send the OPTIONS call?  I just want to make sure I understand :)

Reply:
Erm..., actually client tries to send the request using authorization header (javascript client running in the browser), but the browser stops it and the browser uses HTTP options request (preflight request) to check if the client is actually allowed to send such request... if the server responds positively (using special CORS headers), the request from the client is allowed and is sent to the server...